rclcpp/multithread_executor/multithread_executor.cpp

```diff
@@ -0,0 +1,188 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <condition_variable>
+#include <memory>
```

Comment:
Apparently you need to `#include <functional>` to get `std::hash`.

Reply:
So you went too far here.  I always go for "include what you use".  In this case, you are using:

* chrono
* functional
* memory
* string
* thread

So you should have includes for all of those.

Reply:
Again, apologies.

In my defense, GitHub makes it pretty difficult to determine which specific lines you want deleted... I'm all for context, but it gets confusing when you really are just talking about one or two lines.